protocol/lib/log/utils.go

```diff
@@ -0,0 +1,37 @@
+package log
+
+import (
+	"errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// InfoLog reports msg as an info level log with specified key vals.
+func InfoLog(ctx sdk.Context, msg string, keyvals ...interface{}) {
+	ctx.Logger().Info(msg, keyvals...)
```

Comment:
is it worth validating that `keyvals` is an even length?

What happens if it's an odd-length? Should we note this in the comment?

Reply:
I think we should adopt the same mentality as the underlying library, which doesn't perform validation. In an odd number case, the underlying library + this function discards the last value. I think Logging/metrics libraries should not be able to affect application behavior, so in this case I think we shouldn't panic.